config/configauth/authenticator.go

```diff
@@ -17,76 +17,69 @@ package configauth
 import (
 	"context"
 	"errors"
-	"io"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/metadata"
+
+	"go.opentelemetry.io/collector/component"
 )
 
 var (
-	errNoOIDCProvided   = errors.New("no OIDC information provided")
 	errMetadataNotFound = errors.New("no request metadata found")
-	defaultAttribute    = "authorization"
 )
 
-// Authenticator will authenticate the incoming request/RPC
+// Authenticator is an Extension that can be used as an authenticator for the configauth.Authentication option.
+// Authenticators are then included as part of OpenTelemetry Collector builds and can be referenced by their
+// names from the Authentication configuration. Each Authenticator is free to define its own behavior and configuration options,
+// but note that the expectations that come as part of Extensions exist here as well. For instance, multiple instances of the same
+// authenticator should be possible to exist under different names.
 type Authenticator interface {
-	io.Closer
-
-	// Authenticate checks whether the given context contains valid auth data. Successfully authenticated calls will always return a nil error and a context with the auth data.
-	Authenticate(context.Context, map[string][]string) (context.Context, error)
+	component.Extension
 
-	// Start will
-	Start(context.Context) error
+	// Authenticate checks whether the given headers map contains valid auth data. Successfully authenticated calls will always return a nil error.
+	// When the authentication fails, an error must be returned and the caller must not retry. This function is typically called from interceptors,
+	// on behalf of receivers, but receivers can still call this directly if the usage of interceptors isn't suitable.
+	Authenticate(ctx context.Context, headers map[string][]string) error
 
 	// UnaryInterceptor is a helper method to provide a gRPC-compatible UnaryInterceptor, typically calling the authenticator's Authenticate method.
-	UnaryInterceptor(context.Context, interface{}, *grpc.UnaryServerInfo, grpc.UnaryHandler) (interface{}, error)
+	UnaryInterceptor(ctx context.Context, req interface{}, srvInfo *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error)
```

Comment:
The `ctx` here is the incoming request context, right? Please document that the interceptor is expected to extract the authentication credentials from the context and authenticate them.

Reply:
It is the incoming RPC context, yes. The `DefaultUnaryInterceptor`/`DefaultStreamInterceptor` indeed extracts the credentials from the headers, but some receivers might have a different authentication method, perhaps by a field in the payload. Therefore, I would leave the "how'' to get the auth data out of this doc here.

Reply:
Can we say "authenticate for example by looking up for the credentials in the incoming RPC request context" or something like that?

Reply:
Added a couple of lines in this direction:

```go
	// While the context is the typical source of authentication data, the interceptor is free to determine where the auth data should come from. For instance, some
	// receivers might implement an interceptor that looks into the payload instead.
```